android/app/src/main/java/org/uwaterloo/subletr/pages/emailverification/EmailVerificationPageView.kt

```diff
@@ -0,0 +1,229 @@
+                contentColor = Color.White,
+            )
+        ) {
+            Text(
+                text = stringResource(id = R.string.verify),
+                color = Color.White,
+                style = MaterialTheme.typography.labelMedium,
+                textAlign = TextAlign.Justify
+            )
+        }
+        Spacer(
+            modifier = Modifier.height(20.dp)
+        )
+        Button(
+            modifier = Modifier
+                .fillMaxWidth(ELEMENT_WIDTH)
+                .height(40.dp),
+            onClick = {},
+            colors = ButtonDefaults.buttonColors(
+                containerColor = buttonBackgroundColor,
+                contentColor = Color.White,
+            )
+        ) {
+            Text(
+                text = stringResource(id = R.string.resend_code),
+                color = Color.Gray,
+                style = MaterialTheme.typography.labelMedium,
+
+            )
+        }
+        Spacer(
+            modifier = Modifier.weight(weight = 13.0f)
+        )
+
+    }}
+
+}
+
+
+@Composable
+fun TextFieldBox(index:Int, focusManager:FocusManager,viewModel: EmailVerificationPageViewModel,uiState: EmailVerificationPageUiState = viewModel.uiStateStream.subscribeAsState(
+    EmailVerificationPageUiState.Loading
+).value) {
+    var text by remember { mutableStateOf(TextFieldValue("")) }
+    var btnGrey = buttonBackgroundColor
+    var sublrPink = subletrPink
+
+    var backgroundColor by remember { mutableStateOf(btnGrey) }
+    var borderColor by remember { mutableStateOf(btnGrey) }
+    if (uiState is EmailVerificationPageUiState.Loaded){
+
+
+    TextField(
+        modifier = Modifier
+            .width(54.dp)
+            .height(56.dp)
+            .wrapContentSize(align = Alignment.Center)
+            .border(BorderStroke(2.dp, borderColor), shape = RoundedCornerShape(5.dp))
+            .onFocusChanged {
+                if (it.isFocused) {
+                    borderColor = sublrPink
+                    backgroundColor = Color.White
+                }
+
+            },
+
+        shape = RoundedCornerShape(size = 5.dp),
+        colors = TextFieldDefaults.colors(
+            unfocusedContainerColor = backgroundColor,
+            focusedContainerColor = backgroundColor,
+            unfocusedIndicatorColor = Color.Transparent,
+            focusedIndicatorColor = Color.Transparent,
+
+
+            ),
+        singleLine = true,
+        textStyle = TextStyle(
+            color = Color.Black,
+            fontFamily = FontFamily.Default,
+            fontWeight = FontWeight.Bold,
+            fontSize = 19.sp,
+            letterSpacing = 0.sp,
+            textAlign = TextAlign.Center
+
+        ),
+        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
+
+        value = uiState.verificationCode[index],
+
+        onValueChange = {
+           if (it.length <= 1 ) {
+               val newCode = uiState.verificationCode.toMutableList()
```

Comment:
Instead of making a new mutable list and all that, you can just map the change into the list with `.map` or in this case, probably `.indexedMap`.